### PR TITLE
Metal - Add support for Sync Interval for Mac and ios

### PR DESCRIPTION
**Pull request by moudgils**
Add support foor more detailed name for SRG Pools to help with debugging

Signed-off-by: moudgils <moudgils@amazon.com>

### CODE REVIEW

File: Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp [hunk 1]
```diff
@@ -43,11 +43,36 @@ namespace Platform
...
+        bool framePresented = false;
+        float presentAfterMinimumDuration = syncInterval / refreshRate;
+        
+#if defined(__MAC_11_0)
+        if(@available(iOS 14.0, macOS 11.0, *))
+        {
+            hasPresentAfterMinimumDurationAPI = [drawable respondsToSelector:@selector(presentAfterMinimumDuration:)];
+            if (hasPresentAfterMinimumDurationAPI && presentAfterMinimumDuration > 0.0f)
+            {
+                [mtlCommandBuffer presentDrawable:drawable afterMinimumDuration:presentAfterMinimumDuration];
```
Comment (rgba16f) [inline Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp:66]: This looks wrong. The selector should match the name of the obj-c function you call.

File: Gems/Atom/RHI/Metal/Code/Source/Platform/iOS/RHI/Metal_RHI_iOS.cpp [hunk 2]
```diff
@@ -29,13 +29,20 @@ namespace Platform
...
+    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval, float refreshRate)
+    {
+        float presentAfterMinimumDuration = syncInterval / refreshRate;
         bool hasPresentAfterMinimumDuration = [drawable respondsToSelector:@selector(presentAfterMinimumDuration:)];
                 
-        if (hasPresentAfterMinimumDuration && syncInterval > 0.0f)
+        if (hasPresentAfterMinimumDuration && hasPresentAfterMinimumDuration > 0.0f)
         {
-            [mtlCommandBuffer presentDrawable:drawable afterMinimumDuration:syncInterval];
+            [mtlCommandBuffer presentDrawable:drawable afterMinimumDuration:presentAfterMinimumDuration];
```
Comment (jeremyong-az) [inline Gems/Atom/RHI/Metal/Code/Source/Platform/iOS/RHI/Metal_RHI_iOS.cpp:45]: Do we need a guard to ensure the iOS version is 10.3+? From apple [docs](https://developer.apple.com/documentation/metal/mtlcommandbuffer/2806849-presentdrawable?language=objc)

Comment (rgba16f) [reply Gems/Atom/RHI/Metal/Code/Source/Platform/iOS/RHI/Metal_RHI_iOS.cpp:45]: O3DE min spec on ios is iOS 13 so this should be safe.

File: Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp (hunk 1 above)
Comment (rgba16f) [reply Gems/Atom/RHI/Metal/Code/Source/Platform/Mac/RHI/Metal_RHI_Mac.cpp:66]: Reading the docs it seems [MTLCommandBuffer presentDrawable: afterMinimumDuration:] is a utility function that forwards to [MTLDrawable presentAfterMinimumDuration] at the correct time.

Please add a comment explaining why you check for one function but call the other.

**Review by rgba16f — APPROVED** [review]

**Review by jeremyong-az — APPROVED** [review]

**Pull request closed without merging**